### PR TITLE
prov/sockets: Coverity scan fix and minor code cleanup

### PR DESCRIPTION
**Pull request by shantonu**
- Added check for NULL value of pe_affinity env var to fix coverity scan issue
- Removed a redundant setting of pe_entry->context
- Removed a redundant calculation of data length in pe_entry

@jithinjosepkl can you please review this?
Signed-off-by: Shantonu Hossain <shantonu.hossain@intel.com>

### CODE REVIEW

**Comment by shefty** [comment]
This probably should have been broken into 2 separate patches.  See inline note

File: prov/sockets/src/sock_progress.c [hunk 1]
```diff
@@ -1429,7 +1429,6 @@ static int sock_pe_process_rx_send(struct sock_pe *pe, struct sock_rx_ctx *rx_ct
 			rx_entry->data = pe_entry->data;
 			rx_entry->ignore = 0;
 			rx_entry->comp = pe_entry->comp;
-			pe_entry->context = rx_entry->context;
```
Comment (shefty) [inline prov/sockets/src/sock_progress.c]: Why was this changed?

The change to setting data_len looks like a valid fix, versus a code cleanup.  The change to context needs more explanation.

Comment (shantonu) [inline prov/sockets/src/sock_progress.c]: Sorry, the description should have been clearer. Updating the description

**Comment by shantonu** [comment]
Divided into two patches and updated the description.

File: prov/sockets/src/sock_progress.c [hunk 2]
```diff
@@ -2464,6 +2463,11 @@ static void sock_thread_set_affinity(char *s)
 	int j, first, last,stride;
 	cpu_set_t mycpuset;
 	pthread_t mythread;
+ 	
+	if(!s) {
+		SOCK_LOG_DBG("Invalid FI_SOCKETS_PE_AFFINITY value\n");
+		return;
+	}
```
Comment (shefty) [inline prov/sockets/src/sock_progress.c]: This check shouldn't be needed.  This comes from fi_var_get_str().  If that call returns success, the return string will not be NULL.

Comment (shefty) [inline prov/sockets/src/sock_progress.c]: The coverity issue deals with the saveptra variable:

CID 100409 (#1 of 1): Explicit null dereferenced (FORWARD_NULL)6. var_deref_model: Passing &saveptra to __strtok_r_1c, which dereferences null saveptra. [show details]

This is not an actual issue.  The checker just can't distinguish the first call to strtok_r, versus a subsequent call. I marked this defect as invalid in coverity.  Hopefully that works to dismiss it.

Comment (shantonu) [inline prov/sockets/src/sock_progress.c]: I also found that the check is not needed but since it was reported by coverity, I added it. Thanks for reporting it in coverity.

Comment (shefty) [inline prov/sockets/src/sock_progress.c]: If you re-read the coverity report, the issue is not the 's' parameter into strtok_r, but the 'saveptra' parameter.  This change would not have fixed the issue.

File: prov/sockets/src/sock_progress.c [hunk 3]
```diff
@@ -1403,10 +1403,10 @@ static int sock_pe_process_rx_send(struct sock_pe *pe, struct sock_rx_ctx *rx_ct
...
 		len += SOCK_CQ_DATA_SIZE;
 	}
-
-	if (pe_entry->done_len == len && !pe_entry->pe.rx.rx_entry) {
-		data_len = pe_entry->msg_hdr.msg_len - len;
-
+	
+	data_len = pe_entry->msg_hdr.msg_len - len;
+	
+	if (pe_entry->done_len == len && !pe_entry->pe.rx.rx_entry) {
```
Comment (shefty) [inline prov/sockets/src/sock_progress.c]: nit - please watch introducing unneeded white space, especially at the end of a line.  Some patch tools will reject patches containing this.

**Comment by shefty** [comment]
Merging  manually
